dali/pipeline/operator/direct_operator.h

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_DIRECT_OPERATOR_H_
+#define DALI_PIPELINE_OPERATOR_DIRECT_OPERATOR_H_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/pipeline/data/tensor_list.h"
+#include "dali/pipeline/operator/op_spec.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/util/backend2workspace_map.h"
+#include "dali/pipeline/util/thread_pool.h"
+#include "dali/pipeline/workspace/workspace.h"
+
+namespace dali {
+
+template <typename Backend>
+std::shared_ptr<TensorList<Backend>> AsTensorList(std::shared_ptr<TensorList<Backend>> input) {
+  return input;
+}
+
+template <typename Backend>
+std::shared_ptr<TensorList<Backend>> AsTensorList(std::shared_ptr<TensorVector<Backend>> input) {
+  // TODO(ksztenderski): Remove copy.
+  auto tl = std::make_shared<TensorList<Backend>>();
+  tl->Copy(*input);
+  return tl;
+}
+
+/**
+ * @brief Direct operator providing eager execution of an operator in Run.
+ */
+template <typename Backend>
+class DLL_PUBLIC DirectOperator {
```

Comment:
```suggestion
class DLL_PUBLIC ImmediateOperator {
```
or
```suggestion
class DLL_PUBLIC EagerOperator {
```
?
"Direct" is doesn't really convey this meaning. It's an merely an abbreviation for "DirectlyCalledOperator" which is lengthy.

Reply:
done, went with `EagerOperator`